sentinel1_reader/sentinel1_burst_slc.py

```diff
@@ -0,0 +1,189 @@
+            RadarGridParameters constructed from class members.
+        '''
+
+        prf = 1 / self.azimuth_time_interval
+
+        length, width = self.shape
+
+        time_delta = datetime.timedelta(days=2)
+        ref_epoch = isce3.core.DateTime(self.sensing_start - time_delta)
+        # sensing start with respect to reference epoch
+        sensing_start = time_delta.total_seconds()
+
+        # init radar grid
+        return isce3.product.RadarGridParameters(sensing_start,
+                                                 self.wavelength,
+                                                 prf,
+                                                 self.starting_range,
+                                                 self.range_pixel_spacing,
+                                                 isce3.core.LookSide.Right,
+                                                 length,
+                                                 width,
+                                                 ref_epoch)
+
+    def to_vrt_file(self, out_path):
+        '''
+        Write burst to VRT file.
+
+        Parameters:
+        -----------
+        out_path : string
+            Path of output VRT file.
+        '''
+        line_offset = self.i_burst * self.shape[0]
+
+        inwidth = self.last_valid_sample - self.first_valid_sample
+        inlength = self.last_valid_line - self.first_valid_line
+        outlength, outwidth = self.shape
+        yoffset = line_offset + self.first_valid_line
+        localyoffset = self.first_valid_line
+        xoffset = self.first_valid_sample
+        gdal_obj = gdal.Open(self.tiff_path, gdal.GA_ReadOnly)
+        fullwidth = gdal_obj.RasterXSize
+        fulllength = gdal_obj.RasterYSize
+
+        # TODO maybe cleaner to write with ElementTree
+        tmpl = f'''<VRTDataset rasterXSize="{outwidth}" rasterYSize="{outlength}">
+    <VRTRasterBand dataType="CFloat32" band="1">
+        <NoDataValue>0.0</NoDataValue>
+        <SimpleSource>
+            <SourceFilename relativeToVRT="1">{self.tiff_path}</SourceFilename>
+            <SourceBand>1</SourceBand>
+            <SourceProperties RasterXSize="{fullwidth}" RasterYSize="{fulllength}" DataType="CInt16"/>
+            <SrcRect xOff="{xoffset}" yOff="{yoffset}" xSize="{inwidth}" ySize="{inlength}"/>
+            <DstRect xOff="{xoffset}" yOff="{localyoffset}" xSize="{inwidth}" ySize="{inlength}"/>
+        </SimpleSource>
+    </VRTRasterBand>
+</VRTDataset>'''
+
+        with open(out_path, 'w') as fid:
+            fid.write(tmpl)
+
+    def get_sensing_mid(self):
+        '''
+        Returns sensing mid as datetime object.
+
+        Returns:
+        --------
+        _ : datetime
+            Sensing mid as datetime object.
+        '''
+        d_seconds = 0.5 * (self.shape[0] - 1) * self.azimuth_time_interval
+        return self.sensing_start + datetime.timedelta(seconds=d_seconds)
+
+    def get_isce3_orbit(self, orbit_dir: str):
```

Comment:
I think it makes sense to load orbit while the XML is read and to make it a member of `Sentinel1BurstSlc`. It'll be highly unlikely that a swath will span multiple orbit files, so repeated calls current function is doing a lot repeated/redundant file reading.

Reply:
I agree that orbit may be a member of sentinel-1 burst SLC. But still you need a method to return the orbit as we will need orbit for workflows as you know.
The extra call to get_orbit did not bother me. actually it's more clear what is going on.